### PR TITLE
Shareable link basic functionality

### PR DESCRIPTION
**Pull request by ajithtunerlabs**

### CODE REVIEW

File: controllers/v1/shareablesController.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  constructor() {
+    super(shareableLinksSchema);
+  }
+
+  static get name() {
+    return "shareables";
+  }
+
+  getLink(req) {
```
Comment (ajithtunerlabs) [inline controllers/v1/shareablesController.js]: generate

File: generics/helpers/slackCommunications.js [hunk 2]
```diff
@@ -162,7 +162,107 @@ const rubricErrorLogs = function (errorMessage) {
   }
 }
 
+const badRequestForNonLoggedInUsers = function (errorMessage) {
```
Comment (aks30) [inline generics/helpers/slackCommunications.js]: Change it to badSharedLinkAccessAttemptAlert. Also, add configuration in env to turn these alerts on or off. @ajithtunerlabs

File: controllers/v1/sharedLinksController.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        shareableData = await database.models.sharedLink.findOne({ linkId: linkId, isActive: true }).lean();
+
+        if (!shareableData) throw { status: 400, message: "No data found for given params." };
+
+        let isChanged = false;
+
+        shareableData.linkViews.forEach(user => {
+          if (user.ip == req.ip) {
+           user.accessedCount++;
```
Comment (aks30) [inline controllers/v1/sharedLinksController.js]: Access count is at link level and not at user level. @ajithtunerlabs

File: models/sharedLinks.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+    name: "sharedLink",
+    schema: {
+        actualURL: "string",
+        linkId: "string",
+        isActive: Boolean,
+        sharedURL: "string",
+        linkViews: Array,
+        userDetails: {
```
Comment (aks30) [inline models/sharedLinks.js]: Add how many total times the link was access count here. @ajithtunerlabs

File: generics/helpers/slackCommunications.js (hunk 2 above)
Comment (aks30) [reply generics/helpers/slackCommunications.js]: Also please share a sample slack error message that will come to me.

**Pull request merged**